assert/assertions.go

```diff
@@ -212,19 +214,26 @@ the problem actually occurred in calling code.*/
 func CallerInfo() []string {
 
 	var pc uintptr
-	var ok bool
 	var file string
 	var line int
 	var name string
 
 	callers := []string{}
-	for i := 0; ; i++ {
-		pc, file, line, ok = runtime.Caller(i)
-		if !ok {
-			// The breaks below failed to terminate the loop, and we ran off the
-			// end of the call stack.
-			break
-		}
+	pcs := make([]uintptr, stackFrameBufferSize)
+	offset := 1
+	n := runtime.Callers(offset, pcs)
+	maybeMore := n == stackFrameBufferSize
```

Comment:
I would move this line, just before the `for`

Reply:
same for offset and pcs

They are not used when return via the `if`

Reply:
`pcs` is filled by `runtime.Callers` before the `if` so I'll keep that there; I'll move the other two lines, though

Reply:
Indeed, thanks for applying my first suggestion